networkx/algorithms/structuralholes.py

```diff
@@ -144,17 +144,15 @@ def redundancy(G, u, v, weight=None):
     # Use Borgatti's simplified formula for unweighted and undirected graphs
     if not G.is_directed() and weight is None:
         for v in nodes:
-            # Effective size is not defined for isolated nodes
-            if len(G[v]) == 0:
-                effective_size[v] = float("nan")
+            if len(G[v]) - list(G[v]).count(v) == 0:
+                effective_size[v] = 0
```

Comment:
How about: (for efficiency)
```python
if len(G[v].keys() - {}) == 0:
```
or maybe (I haven't check the timing)
```python
if len(G[v]) - (v in G[v]) == 0:
```

Reply:
Thanks for that! I am not sure about the MultiGraph that contains multiple edges `<v, v>`, how does the behavior of these codes? I will try to check them both for functionalities and timing.

Best regards,
Joye

Reply:
MultiGraph has the structure node-dict of nbr-dict of edgekey-dict of edgeattribute-dict.  So `G[u]` is a dict keyed by nbr. And `G[n][nbr]` is a dict keyed by edgekey, so `len(G[n][nbr])` is the number of edges between `n` and `nbr`. For this use, the idiom `v in G[v]` returns `True` when `v` is a neighbor of `v`. And that works for either Graph or MultiGraph.  Generally we have tried to arrange it so most code written for Graph can work as expected for MultiGraph. That doesn't always work, and certainly doesn't work the other way around. :)

Reply:
It makes sense, I will try to give a code that also works well on the MultiGraph